Update DomainEntity_EmailUrlInfo.yaml

MSJosh opened the pull request:
Removed materialize as it causes a memory issue on larger data sets.  Improved performance of query by splitting the logic.

   Required items, please complete
   
   Change(s):
   -Removed materialize() functions
   -Split single query into two different queries to improve performance
   -Cleaned up sub queries

   Reason for Change(s):
   - For larger data sets this would lead to memory issues and would fail to run or lead to shorter look back periods.
   - Resolves Issue 11340

   Version Updated:
   - 1.0.2

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

MSJosh commented:
@microsoft-github-policy-service agree company="Microsoft"

________________________________
From: microsoft-github-policy-service[bot] ***@***.***>
Sent: Thursday, October 31, 2024 12:20 PM
To: Azure/Azure-Sentinel ***@***.***>
Cc: Josh Kolka ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] Update DomainEntity_EmailUrlInfo.yaml (PR #11369)


@MSJosh<https://github.com/MSJosh> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.

@microsoft-github-policy-service agree [company="{your company}"]


Options:

  *   (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.

@microsoft-github-policy-service agree


  *   (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.

@microsoft-github-policy-service agree company="Microsoft"


Contributor License Agreement
Contribution License Agreement

This Contribution License Agreement (“Agreement”) is agreed to by the party signing below (“You”),
and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your
contributions to Microsoft open source projects. This Agreement is effective as of the latest signature
date below.

  1.  Definitions.
“Code” means the computer software code, whether in human-readable or machine-executable form,
that is delivered by You to Microsoft under this Agreement.
“Project” means any of the projects owned or managed by Microsoft and offered under a license
approved by the Open Source Initiative (www.opensource.org<http://www.opensource.org/>).
“Submit” is the act of uploading, submitting, transmitting, or distributing code or other content to any
Project, including but not limited to communication on electronic mailing lists, source code control
systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
discussing and improving that Project, but excluding communication that is conspicuously marked or
otherwise designated in writing by You as “Not a Submission.”
“Submission” means the Code and any other copyrightable material Submitted by You, including any
associated comments and documentation.
  2.  Your Submission. You must agree to the terms of this Agreement before making a Submission to any
Project. This Agreement covers any and all Submissions that You, now or in the future (except as
described in Section 4 below), Submit to any Project.
  3.  Originality of Work. You represent that each of Your Submissions is entirely Your original work.
Should You wish to Submit materials that are not Your original work, You may Submit them separately
to the Project if You (a) retain all copyright and license information that was in the materials as You
received them, (b) in the description accompanying Your Submission, include the phrase “Submission
containing materials of a third party:” followed by the names of the third party and any licenses or other
restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
guidelines concerning Submissions.
  4.  Your Employer. References to “employer” in this Agreement include Your employer or anyone else
for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
Submission is made in the course of Your work for an employer or Your employer has intellectual
property rights in Your Submission by contract or applicable law, You must secure permission from Your
employer to make the Submission before signing this Agreement. In that case, the term “You” in this
Agreement will refer to You and the employer collectively. If You change employers in the future and
desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
and secure permission from the new employer before Submitting those Submissions.
  5.  Licenses.

  *   Copyright License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
parties.
  *   Patent License. You grant Microsoft, and those who receive the Submission directly or
indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
Your patent claims that are necessarily infringed by the Submission or the combination of the
Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
import or otherwise dispose of the Submission alone or with the Project.
  *   Other Rights Reserved. Each party reserves all rights not expressly granted in this Agreement.
No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
granted by implication, exhaustion, estoppel or otherwise.

  1.  Representations and Warranties. You represent that You are legally entitled to grant the above
licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
have disclosed under Section 3). You represent that You have secured permission from Your employer to
make the Submission in cases where Your Submission is made in the course of Your work for Your
employer or Your employer has intellectual property rights in Your Submission by contract or applicable
law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
  2.  Notice to Microsoft. You agree to notify Microsoft in writing of any facts or circumstances of which
You later become aware that would make Your representations in this Agreement inaccurate in any
respect.
  3.  Information about Submissions. You agree that contributions to Projects and information about
contributions may be maintained indefinitely and disclosed publicly, including Your name and other
information that You submit with Your Submission.
  4.  Governing Law/Jurisdiction. This Agreement is governed by the laws of the State of Washington, and
the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
defenses of lack of personal jurisdiction and forum non-conveniens.
  5.  Entire Agreement/Assignment. This Agreement is the entire agreement between the parties, and
supersedes any and all prior agreements, understandings or communications, written or oral, between
the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/Azure-Sentinel/pull/11369#issuecomment-2450292429>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A4XLFSIJNIDPMGM7TEF6VNDZ6JKFXAVCNFSM6AAAAABQ6UYJ6OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINJQGI4TENBSHE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

v-prasadboke commented:
Hi Josh, 

I saw your mail as you said it worked on alpine ski house 
But can you check and fix this if possible

v-prasadboke commented:
Hello @MSJosh, Just discussed about this internally. 

We need to use the template which is mentioned in the validation check 

   **Errors: Content needs to use the latest Threat Intelligence data. Sample queries to get the latest Threat Intelligence data:
        ThreatIntelligenceIndicator
        | where TimeGenerated >= ago(ioc_lookBack)
        | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
        | where Active == true and ExpirationDateTime > now()

        or

        ThreatIntelligenceIndicator
        | where TimeGenerated >= ago(ioc_lookBack)
        | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
        | where ExpirationDateTime > now() and Active == true**

Reason being, above query fetches the latest TI data or logs. 
Neglecting this would fetch the expired data or false data.

MSJosh commented:
Not understanding what you need me to do.  Are you wanting to modify the code to put the
| summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
| where Active == true and ExpirationDateTime > now()

In the portions where I have wrote for TI??

________________________________
From: v-prasadboke ***@***.***>
Sent: Friday, November 15, 2024 5:58 AM
To: Azure/Azure-Sentinel ***@***.***>
Cc: Josh Kolka ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/Azure-Sentinel] Update DomainEntity_EmailUrlInfo.yaml (PR #11369)


Hello @MSJosh<https://github.com/MSJosh>, Just discussed about this internally.

We need to use the template which is mentioned in the validation check

**Errors: Content needs to use the latest Threat Intelligence data. Sample queries to get the latest Threat Intelligence data:
ThreatIntelligenceIndicator
| where TimeGenerated >= ago(ioc_lookBack)
| summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
| where Active == true and ExpirationDateTime > now()

    or

    ThreatIntelligenceIndicator
    | where TimeGenerated >= ago(ioc_lookBack)
    | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
    | where ExpirationDateTime > now() and Active == true**


Reason being, above query fetches the latest TI data or logs.
Neglecting this would fetch the expired data or false data.

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/Azure-Sentinel/pull/11369#issuecomment-2478560277>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A4XLFSM4ZQYTD4KK3BO43EL2AXHW5AVCNFSM6AAAAABQ6UYJ6OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZYGU3DAMRXG4>.
You are receiving this because you were mentioned.

v-prasadboke commented:
> Not understanding what you need me to do. Are you wanting to modify the code to put the | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId | where Active == true and ExpirationDateTime > now() In the portions where I have wrote for TI??
> […](#)
> ________________________________ From: v-prasadboke ***@***.***> Sent: Friday, November 15, 2024 5:58 AM To: Azure/Azure-Sentinel ***@***.***> Cc: Josh Kolka ***@***.***>; Mention ***@***.***> Subject: Re: [Azure/Azure-Sentinel] Update DomainEntity_EmailUrlInfo.yaml (PR #11369) Hello @MSJosh<https://github.com/MSJosh>, Just discussed about this internally. We need to use the template which is mentioned in the validation check **Errors: Content needs to use the latest Threat Intelligence data. Sample queries to get the latest Threat Intelligence data: ThreatIntelligenceIndicator | where TimeGenerated >= ago(ioc_lookBack) | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId | where Active == true and ExpirationDateTime > now() or ThreatIntelligenceIndicator | where TimeGenerated >= ago(ioc_lookBack) | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId | where ExpirationDateTime > now() and Active == true** Reason being, above query fetches the latest TI data or logs. Neglecting this would fetch the expired data or false data. — Reply to this email directly, view it on GitHub<[#11369 (comment)](https://github.com/Azure/Azure-Sentinel/pull/11369#issuecomment-2478560277)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A4XLFSM4ZQYTD4KK3BO43EL2AXHW5AVCNFSM6AAAAABQ6UYJ6OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINZYGU3DAMRXG4>. You are receiving this because you were mentioned.

Yes

MSJosh commented:
@v-prasadboke Looks like I have everything fixed minus one item. I did a new pull to fix the one issue found. I fixed the issue in pull request 11468, where I added Workday as a source as the team didn't have it added as part of the new CCP solution for Workday.

MSJosh reviewed (commented):
Please run KQL validation against this. Looks like it was an issue with the verifier.

MSJosh reviewed (commented):
Running verifier again

v-prasadboke commented:
Hello @MSJosh, Can you package this solution once 
I tried to pull commits to my local but I'm facing some issue

Please package the solution using V3 tool once. 
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

MSJosh commented:
@v-prasadboke please take a look and validate. There should be a new build 3.09 available in this latest build. The arm-ttk check is flagging but has nothing to do with the changes made for these detections and looks like they are historical configs that have existed for some time.

MSJosh commented:
Other commit went live and created a fresh pull for workday to keep it clean.

v-prasadboke commented:
Thanks for the update Josh
